src/EnergyPlus/DataHeatBalance.cc

```diff
@@ -1294,71 +1294,62 @@ Real64 ComputeNominalUwithConvCoeffs(EnergyPlusData &state,
+    // is connected to a semi-exterior area of the building.  Users can always use
+    // the Nominal U-Value to manually calculated this.  The values calculated here
+    // are simply reported to the EIO file and not used for any calculations.
 
     // Return value
     Real64 NominalUwithConvCoeffs; // return value
 
+    static constexpr std::array<Real64, static_cast<int>(DataSurfaces::SurfaceClass::Num)> filmCoefs = {
+        0.0,       // None
+        0.1197548, // Wall
+        0.1620212, // Floor
+        0.1074271, // Roof
+        0.0,       // IntMass
+        0.0,       // Detached_B
+        0.0,       // Detached_F
+        0.1197584, // Window
+        0.1197584, // GlassDoor
+        0.1197584, // Door
+        0.0,       // Shading
+        0.0,       // Overhang
+        0.0,       // Fin
+        0.0,       // TDD_Dome
+        0.0        // TDD_Diffuser
+    };
+
     Real64 insideFilm;
     Real64 outsideFilm;
 
     isValid = true;
+
+    auto &thisSurface = state.dataSurface->Surface(numSurf);
+
     // exterior conditions
-    switch (state.dataSurface->Surface(numSurf).ExtBoundCond) {
-    case ExternalEnvironment: {
-        outsideFilm = 0.0299387; // All exterior conditions
+    switch (thisSurface.ExtBoundCond) {
+    case DataSurfaces::ExternalEnvironment: { // ExtBoundCond = 0
+        outsideFilm = 0.0299387;              // All exterior conditions
     } break;
-    case Ground:
-    case GroundFCfactorMethod: {
-        outsideFilm = 0.0; // No outside film when underground
+    case DataSurfaces::Ground:
+    case DataSurfaces::OtherSideCoefNoCalcExt:
+    case DataSurfaces::OtherSideCoefCalcExt:
+    case DataSurfaces::OtherSideCondModeledExt:
+    case DataSurfaces::GroundFCfactorMethod:
+    case DataSurfaces::KivaFoundation: { // All these cases have a negative ExtBoundCond so don't use film coefficients
+        outsideFilm = 0.0;
     } break;
-    default: {
-        if (state.dataSurface->Surface(numSurf).ExtBoundCond > 0) { // interzone partition
-            // use companion surface in adjacent zone
-            switch (state.dataSurface->Surface(state.dataSurface->Surface(numSurf).ExtBoundCond).Class) {
-            case SurfaceClass::Wall:
-            case SurfaceClass::Door: { // Interior:  vertical, still air, Rcin = 0.68 ft2-F-hr/BTU
-                outsideFilm = 0.1197548;
-            } break;
-            case SurfaceClass::Floor: { // Interior:  horizontal, still air, heat flow downward, Rcin = 0.92 ft2-F-hr/BTU
-                outsideFilm = 0.1620212;
-            } break;
-            case SurfaceClass::Roof: { // Interior:  horizontal, still air, heat flow upward, Rcin = 0.61 ft2-F-hr/BTU
-                outsideFilm = 0.1074271;
-            } break;
-            default: {
-                outsideFilm = 0.0810106; // All semi-exterior surfaces
-            } break;
-            }
-        } else {
-            outsideFilm = 0.0810106; // All semi-exterior surfaces
```

Comment:
Where did this semi-exterior coefficient (0.0810106) get moved to? or was it never used?

Reply:
It should never have been used (was used in error) and was removed from the code.  There's nothing in the input that would indicate definitely a semi-exterior surface.  It was used as a default (incorrectly) before.